content/tutorials/reactor-event-ifttt.textile

```diff
@@ -0,0 +1,196 @@
+# In the resulting form, under "Which channnel":
+## select Direct Messages 
+## select your own Slack handle
+# Under "Message":
+## delete the content in the textbox 
+## Click "*Add ingredient*"
+## Add "Value1"
+# Under "Title URL" add "Captain Picard" (or the captain name of your choice!)
+# You can even add a thumbnail avatar of your captain if you'd like by pasting a url to your chosen image
+# Click "*Create Action*"
+# Click "*Finish*"
+
+You have now set up your Ably channel to post to Slack using IFTTT! 
+
+In the next section we'll create a game which will publish messages to our Ably channel, thereby triggering a Slack message to be sent.
+
+h2. Building the Game
+
+To get you up and running quickly, we've made the code for the game available via "Glitch":https://glitch.com/~ably-space-game. You can remix or clone this repository. Press the *Remix to Edit* button in the top right hand corner of the page. We'll need to edit the code a little to add in your Ably API key.
+
+h2. How the game works
+
+The game is a web app, created with "Glitch":https://glitch.com/. It uses "Node":https://nodejs.org/en/ and "Express":https://expressjs.com/. Instructions on how the game works can be read at in the "readme.md":https://glitch.com/edit/#!/ably-space-game?path=README.md:1:0. We're using the game to generate data to send to our Ably channel, if you already have data going to an Ably channel then you will not need the game at all! The game generates 10 'moves' for the player to take and publishes them, one by one, to an Ably channel as the player completes previous moves.
+
+h2. How does the game publish messages?
+
+The game publishes message "commands" to an Ably channel. Take a look at the *ablyConnector.js* file in the root, this uses the Ably REST API to publish to an Ably channel. You'll see some code like this
+
+```js
+const apiUrl = "https://rest.ably.io/channels/space-game/messages?key=" + process.env.ABLY_API_KEY;
+const jsonBody = {"data": {value1: status.hint}};
+
+
+function sendToApi(jsonBody) {
+    try {
+      await fetch(apiUrl, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(jsonBody)
+      });
+
+      console.log("Sent to Ably");
+    } catch (error) {
+      console.log(error);
+    }
+}...
+```
+
+This code publishes `status.hint` (which is our move command) to a channel called "space-game"
+
+The important part to note here is that *IFTTT requires the data it receives to be in the following JSON format*:
+
+```
+{"value1":"foo", "value2":"bar", "value3":"baz"}
+```
+Without this format, it is still possible to trigger an event, but not to send cusomised messages. In the case of this game, we'll be sending only value1, which will contain the move command.
+
+h2. Configuring the game
+
+In order to publish messages, the game will need to have your Ably API key. We store this in a *.env* file to keep it secure. Do not commit your *.env* files to github.
```

Comment:
> Do not commit your *.env* files to github.

I do not think this is a good instruction for beginner developers.  I think we should tell them to add an exclusion to their `.gitconfig` in that project, or in their global config, and provide a link to find out more.  Telling them not to commit the file is error prone as later they will mistakenly.  In addition, we should be mentioning that in practice, we would not recommend clients use API key auth, and we recommend token auth.

Reply:
Adding the exclusion is done by default in Glitch, so if they're remixing the repo this is already done for them, this was more a throwaway comment, I can remove it?

Reply:
> Adding the exclusion is done by default in Glitch

Sure, but are we saying the demo only works on Glitch then?  Up to you, not a biggie, just a passing comment really.  Perhaps just go with "Do not commit your *.env* files to github, generally adding an exclusion in your .gitignore file will make sure of that."